app/javascript/components/patient/laboratory/Laboratory.js

```diff
@@ -1,30 +1,26 @@
 import React from 'react';
 import { PropTypes } from 'prop-types';
-import { Button } from 'react-bootstrap';
+import { Button, Dropdown } from 'react-bootstrap';
 import axios from 'axios';
-
 import LaboratoryModal from './LaboratoryModal';
 import reportError from '../../util/ReportError';
+import DeleteDialog from '../../util/DeleteDialog';
 
 class Laboratory extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      showModal: false,
+      showLabModal: false,
+      showDeleteModal: false,
       loading: false,
     };
   }
 
-  toggleModal = () => {
-    this.setState(state => {
-      return {
-        showModal: !state.showModal,
-        loading: false,
-      };
-    });
+  handleChange = event => {
+    this.setState({ [event.target.id]: event.target.value });
   };
 
-  submit = lab => {
+  handleLabSubmit = lab => {
```

Comment:
this is _super_ nitpicky but can we keep the lab methods together and delete together? something like lab toggle/submit then delete toggle/submit?